test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/AbstractLocalClusterFactory.java

```diff
@@ -476,12 +486,19 @@ private void addKeystoreFiles() {
         }
 
         private void writeSecureSecretsFile() {
-            if (spec.getSecrets().isEmpty() == false) {
+            if (spec.getKeystoreFiles().isEmpty() == false) {
+                throw new IllegalStateException(
+                    "Non-string secure secrets are not supported in serverless. Secrets: ["
+                        + spec.getKeystoreFiles().keySet().stream().collect(Collectors.joining(","))
+                        + "]"
+                );
+            }
+            if (spec.resolveKeystore().isEmpty() == false) {
                 try {
                     Path secretsFile = configDir.resolve("secrets/secrets.json");
                     Files.createDirectories(secretsFile.getParent());
                     Map<String, Object> secretsFileContent = new HashMap<>();
-                    secretsFileContent.put("secrets", spec.getSecrets());
+                    secretsFileContent.put("secrets", spec.resolveKeystore());
```

Comment:
It would be better not to resolve the keystore contents twice twice. Could we assign it to a local var?

Reply:
Done.